app/browser/reducers/passwordManagerReducer.js

```diff
@@ -86,178 +87,145 @@ const init = () => {
+                                               masterKey,
+                                               password.get('iv'))
+      let form = {}
+      form['origin'] = password.get('origin')
+      form['signon_realm'] = password.get('origin') + '/'
+      form['action'] = password.get('action')
+      form['username'] = password.get('username')
+      form['password'] = decrypted
+      autofill.addLogin(form)
     })
-    if (result && password === CryptoUtil.decryptVerify(result.get('encryptedPassword'),
-                                                        result.get('authTag'),
-                                                        masterKey,
-                                                        result.get('iv'))) {
-      return
-    }
-
-    var message = username
-      ? locale.translation('notificationPasswordWithUserName').replace(/{{\s*username\s*}}/, username).replace(/{{\s*origin\s*}}/, origin)
-      : locale.translation('notificationPassword').replace(/{{\s*origin\s*}}/, origin)
-
-    if (!(message in passwordCallbacks)) {
-      // Notification not shown already
-      appActions.showNotification({
-        buttons: [
-          {text: locale.translation('yes')},
-          {text: locale.translation('no')},
-          {text: locale.translation('neverForThisSite')}
-        ],
-        options: {
-          persist: false,
-          advancedText: locale.translation('notificationPasswordSettings'),
-          advancedLink: 'about:passwords'
-        },
-        message
-      })
-    }
-
-    passwordCallbacks[message] = (buttonIndex) => {
-      delete passwordCallbacks[message]
-      appActions.hideNotification(message)
-
-      if (buttonIndex === 1) {
-        return
-      }
-      if (buttonIndex === 2) {
-        // Never save the password on this site
-        appActions.changeSiteSetting(origin, 'savePasswords', false)
-        return
-      }
-
-      // Save the password
-      const encrypted = CryptoUtil.encryptAuthenticate(password, masterKey)
-      appActions.savePassword({
-        origin,
-        action,
-        username: username || '',
-        encryptedPassword: encrypted.content,
-        authTag: encrypted.tag,
-        iv: encrypted.iv
-      })
-    }
-  })
+    state = state.set('passwords', new Immutable.List())
+  }
+  const allSiteSettings = state.get('siteSettings')
+  const blackedList = allSiteSettings.filter((setting) => setting.get('savePasswords') === false)
+  if (blackedList.size) {
+    blackedList.forEach((entry, index) => {
+      let form = {}
+      form['origin'] = index
+      form['signon_realm'] = index + '/'
+      form['blacklisted_by_user'] = true
+      autofill.addLogin(form)
+      appActions.deletePasswordSite(index)
+    })
+  }
+  return state
 }
 
 const passwordManagerReducer = (state, action) => {
   action = makeImmutable(action)
   switch (action.get('actionType')) {
     case appConstants.APP_SET_STATE:
       init()
+      state = migrate(state)
```

Comment:
@bridiver I tried to do it in sessionStore at the migration section but our migration requires `electron.session`. And at that time app is not ready so we will have `Error: Session can only be received when app is ready`